src/transport/FabricTable.cpp

```diff
@@ -412,9 +410,7 @@ CHIP_ERROR FabricInfo::VerifyCredentials(const ByteSpan & noc, ValidationContext
         }
     }
 
-    Encoding::LittleEndian::BufferWriter bbuf(nocPubkey, nocPubkey.Length());
-    bbuf.Put(certificates.GetCertSet()[nocCertIndex].mPublicKey.data(), certificates.GetCertSet()[nocCertIndex].mPublicKey.size());
-    VerifyOrReturnError(bbuf.Fit(), CHIP_ERROR_BUFFER_TOO_SMALL);
+    nocPubkey = P256PublicKey(certificates.GetCertSet()[nocCertIndex].mPublicKey);
```

Comment:
How is nocPubkey used from the output of this function. nocPubkey is not something that should be needed anywhere to be cached/used.

Also, `GetCertSet()` possibly loads the whole key chain, which is costly. Could it be avoided?

Reply:
nocPubkey is used by CASE state machine as `responderID`.
I'll look at optimizing out `GetCertSet()`.

Reply:
As we together looked in the code, `GetCertSet()` is not loading the cert chain. It's returning the cert set which is already loaded.